src/helm/proxy/tokenizers/test_huggingface_tokenizer.py

```diff
@@ -59,6 +60,24 @@ def test_decode(self):
         assert result.cached, "Result should be cached"
         assert result.text == "I am a computer scientist."
 
+    def test_already_borrowed(self):
+        """Demonstrates the "Already borrowed" bug (#1421) caused by the thread-hostile Hugging Face tokenizer"""
+
+        def make_tokenize_request(seed: int) -> None:
+            request_length = 100
+            truncation = bool(seed % 2)
+            self.tokenizer.tokenize(
+                # The truncation parameter requires setting a flag on the Rust FastTokenizer.
+                # Concurrent requests cause concurrent mutations, which results an Rust concurrency error.
+                TokenizationRequest(
+                    text=str(seed) * request_length, tokenizer="huggingface/gpt2", encode=True, truncation=truncation
+                )
+            )
+
+        with pytest.raises(ValueError, match="Already borrowed"):
+            num_requests = 1000
```

Comment:
How long does this test take to run?  What's the smallest number of requests that surfaces this bug?

Reply:
I ran this twenty times with `request_length = 10` and `num_requests = 100` and it reliably hit the bug on all twenty out of twenty runs.

When the bug exists in main, this code terminates on the first exception and skips the rest of the requests; it takes about 0.15 seconds to terminate.

When the bug is fixed in #2091, this code runs all of the requests; it takes about 4 seconds on `request_length = 100` and `num_requests = 1000` and about 0.4 seconds on `request_length = 10` and `num_requests = 100`.

I chose to use the smaller set of parameters (`request_length = 10` and `num_requests = 100`) in the bugfix PR #2091.

Reply:
Also noting that it typically raises the exception within the first 10 requests on my 20 test runs.